src/main/java/org/apache/commons/lang3/text/StrBuilder.java

```diff
@@ -189,8 +198,9 @@ public StrBuilder setNullText(String nullText) {
      *
      * @return the length
      */
+    @SuppressWarnings("override.return.invalid") // it says required is @LTEqLengthOf("this"), which does not makes sense, it should be @LTEqLengthOf("this.buffer")
     @Override
-    public int length() {
+    public @NonNegative @LTEqLengthOf("this.buffer") int length() {
```

Comment:
You're right that a warning needs to be suppressed here, but the annotations are wrong. This class is a `CharSequence` implementation, so a programmer should be able to use it like any other `CharSequence`: that is, they shouldn't be exposed to its implementation details.

The correct signature is `@IndexOrHigh("this")`.

You'll have to suppress a warning to connect the implementation detail `this.buffer` to the `this` object. I suggest doing so on a `SameLen` annotation on `this.buffer`, as described here: https://checkerframework.org/manual/#index-annotating-fixed-size